pkg/runtime/reconciler.go

```diff
@@ -244,18 +248,9 @@ func (r *resourceReconciler) Sync(
 	// Attempt to late initialize the resource. If there are no fields to
 	// late initialize, this operation will be a no-op.
 	if latest, err = r.lateInitializeResource(ctx, rm, latest); err != nil {
-		// TODO(vijtrip2): move this condition handling to generated
-		// AWSResourceManager.LateInitialize() method
-
-		// Whenever late initialization fails for a resource, set ACK.ResourceSynced
-		// condition explicitly to "False"
-		// Setting this explicitly to False is required because ACK.ResourceSynced
-		// condition can be True due to successful Create/Update call OR no
-		// Create/Update call in reconciler loop
-		ackcondition.SetSynced(latest, corev1.ConditionFalse, nil, nil)
 		return latest, err
 	}
-	return r.handleRequeues(ctx, latest)
+	return latest, nil
```

Comment:
nit:
```suggestion
	return r.lateInitializeResource(ctx, rm, latest)
```

Reply:
This was intentional because we have defer exit(err) call in Sync function. So I wanted to assign the error into "err" variable.

Reply:
Make sense! Thanks for clarifying @vijtrip2

Reply:
The call to `exit(err)` doesn't actually work correctly anyway, because of this:

https://stackoverflow.com/questions/42703707/when-defer-func-evaluates-its-parameters/42703862#42703862

We need to change the `exit` function to take `*error` instead of `error`.

Reply:
Most of the errors returned by error packages are pointers e.g: https://github.com/aws/aws-sdk-go/blob/1f3b1b4bd14f4187976f81ea5bea1f9279a57122/aws/awserr/types.go#L50, and `error` type is an interface. I'm not sure if `*error` will work but curious to see it in action